lib/files/node.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * @copyright Copyright (c) 2022 John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @author John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+import { basename, extname, dirname } from 'path'
+import { Permission } from '../permissions'
+import NodeData, { Attribute, validateData } from './nodeData'
+
+const davService = 'remote.php/dav'
```

Comment:
We also have `public.php/webdav` for public shares. Wouldn't hard-coding this be a problem ?

Reply:
Very good point!
I could have a bit wider selector. Or allow modifying it yes :thinking:

Reply:
Done, I added a way to customized it and supported our multiple services
https://github.com/nextcloud/nextcloud-files/blob/b4ae6b156c3f9ff0170d78985bf7b889aa3e7f83/lib/files/node.ts#L27

Tests added too